### PR TITLE
[DC-2402] - Extend user frontend with supporting more enrolments/tax ids

### PR DESCRIPTION
**Pull request by elabeca**

### CODE REVIEW

**Comment by elabeca** [comment]
This is the second part PR (frontend part). Will finalise with more tests.

File: app/controllers/EnquiryController.scala [hunk 1]
```diff
@@ -44,30 +44,38 @@ class EnquiryController @Inject()(appConfig: AppConfig,
...
-          } yield {
-            submissionDetails match {
-              case Some(details) => Ok(enquiry(appConfig, form, EnquiryDetails(enquiryType, "", "", email), details.responseTime))
-              case None => NotFound
-            }
-          }
-        case _ => Future.successful(Forbidden)
-      } recoverWith {
-        case _ : AuthorisationException => Future.successful(Unauthorized)
+    implicit request => {
```
Comment (lxol) [inline app/controllers/EnquiryController.scala:49]: I think if delegate all enquiryType concerns to two-way-message we shouldn't check it on two-way-message-frontend. Personally I would check 
authorised(AuthProvider(GovernmentGateway))
and that's it.
twoWayMessageConnector.getSubmissionDetails(enquiryType) would give us all the checks that we need

Comment (elabeca) [reply app/controllers/EnquiryController.scala:49]: Initially it was but following a discussion with @dominic-messenger-hmrc, Michael and Stephen it was decided to keep the P800 functionality as-is and untouched. That said, I will update the catch all for authorised(AuthProvider(GovernmentGateway)) but I have to keep the special case for P800 under the clamping of HMRC-NI

Comment (elabeca) [reply app/controllers/EnquiryController.scala:49]: I could move the functionality of clamping down to HMRC-NI for P800 only in two-way-message getEnquiryTypeDetails and createMessage?

Comment (elabeca) [reply app/controllers/EnquiryController.scala:49]: As discussed on standup today, it seems that case for P800 is already handled by the AuthIdentifiersConnector in two-way-message. I will update this PR to authorised(AuthProvider(GovernmentGateway)). thank you @lxol

Comment (elabeca) [reply app/controllers/EnquiryController.scala:49]: @lxol please note I still need to get the nino to pre-fill the email address for P800 so this logic needs to remain. But I've removed that logic elsewhere.

**Pull request closed without merging**